Add Regulation class and conformsTo relationship

colin-pm opened the pull request:
This PR intends to add support for a conformsTo relationship and Regulation class. The goal is to be able to define relationships between software packages and regulations, for example.

colin-pm committed:
Add Regulation

Adds a new regulation class that can be used to define rules and
directives.

Signed-off-by: Colin Pinnell McAllister <colinmca242@gmail.com>

colin-pm committed:
Add conformsTo relationship

Adds confromsTo relationship that's intended to be used with new
Regulation class.

Signed-off-by: Colin Pinnell McAllister <colinmca242@gmail.com>

bact commented:
@stevenc-stb this could be related to Standard - and Specification we have discussed in Tech and AI Team meetings

bact commented on model/Core/Classes/Regulation.md:
```diff
@@ -0,0 +1,19 @@
+
+## Description
+
+The Regulation class represents any rules or directives enforced by a central authority.
```
```suggestion
The Regulation class represents any rules or directives enforced by a competent authority. Regulations within this class can be either mandatory or voluntary.
```

bact commented on model/Core/Classes/Regulation.md:
```diff
@@ -0,0 +1,19 @@
+
+The Regulation class represents any rules or directives enforced by a central authority.
+
+This can be regulations such as EU RED, EU CRA, or the U.S. Cyber Trust Mark.
```
```suggestion
Examples of these regulations include the EU Radio Equipment Directive, the EU Cyber Resilience Act, the Japan Cyber STAR, or the U.S. Cyber Trust Mark.
```

Expand RED and CRA, and also add a non-Western example.

bact reviewed (commented):
Add few suggestions, feel free to adjust.

1) Central authority:
    - "central authority" could bring uncertainty if a local or non-federal level authority applies here.
    - If excluding regulations from local authorities is not the intention, I suggest to replace "central authority" with "competent authority" or "official authority".
2) Voluntary included?:
    - Since we have U.S. Cyber Trust Mark in the example, I guess the intention of this Regulation class includes regulations of both mandatory and voluntary nature.
    - It would be useful to explicitly mention that in the description as well.
3) Example:
    - Expands EU RED and EU CRA.
    - Add a non-Western example to the list.

Do you like to attend the Tech call on Tuesday to discuss this?
See the time/link here: https://github.com/spdx/meetings/?tab=readme-ov-file#tech-team-meetings

colin-pm commented:
> Add few suggestions, feel free to adjust.
> 
> 1. Central authority:
>    
>    * "central authority" could bring uncertainty if a local or non-federal level authority applies here.
>    * If excluding regulations from local authorities is not the intention, I suggest to replace "central authority" with "competent authority" or "official authority".

"official authority" sounds good to me.

> 2. Voluntary included?:
>    
>    * Since we have U.S. Cyber Trust Mark in the example, I guess the intention of this Regulation class includes regulations of both mandatory and voluntary nature.
>    * It would be useful to explicitly mention that in the description as well.

I added this line. Since the regulations could be mandatory or voluntary, I added a mandatory bool property.

> 3. Example:
>    
>    * Expands EU RED and EU CRA.
>    * Add a non-Western example to the list.

Made these changes, thank you.

> Do you like to attend the Tech call on Tuesday to discuss this? See the time/link here: https://github.com/spdx/meetings/?tab=readme-ov-file#tech-team-meetings

I'd be happy to join if you think there is more to discuss. Thanks for the review!

bact commented on model/Core/Classes/Regulation.md:
```diff
@@ -0,0 +1,28 @@
+## Properties
+
+- mandatory
+  - Range: xsd:boolean
```
```suggestion
  - type: xsd:boolean
```

This should be "type".
See https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#classes

bact commented:
@colin-pm I see irrelevant changes in two files below that may be added accidentally? Would you mind to remove them from the changeset please? Thank you.

- serialization/json.md
- serialization/jsonld/validation.md

bact commented:
> >    * Since we have U.S. Cyber Trust Mark in the example, I guess the intention of this Regulation class includes regulations of both mandatory and voluntary nature.
> >    * It would be useful to explicitly mention that in the description as well.
> 
> I added this line. Since the regulations could be mandatory or voluntary, I added a mandatory bool property.
 
There is one error in the validation (see the [run log](https://github.com/spdx/spdx-3-model/actions/runs/14303766239/job/40082976188?pr=1015)) since `mandatory` property is added and it is not existed anywhere yet in the current model.
To use this property, you have to introduce this property to the model as well. Probably in model/Core/Properties.

> I'd be happy to join if you think there is more to discuss. Thanks for the review!

There are discussions about `Standard`/`Specification` class, which may be related to your proposal of `Regulation` and `conformsTo` (A conformsTo Specification, for example).
See AI Team Meeting 2025-04-02: https://spdx.swinslow.net/p/spdx-ai-minutes ("Hardware Profile and common fields"):

> - Standard class - has been abstracted to be a type of Artifact, rather than just independent field.
>   - Independent field is also being used in operations
>   - We have 2 "standard" fields in 3.0 today (https://github.com/spdx/spdx-3-model/issues/893):
>     In Core https://github.com/spdx/spdx-3-model/blob/develop/model/Core/Properties/standardName.md
>     In AI [https://github.com/spdx/spdx-3-model/blob/develop/model/AI/Properties/standardCompliance](https://github.com/spdx/spdx-3-model/blob/develop/model/AI/Properties/standardCompliance.md)[.md](https://github.com/spdx/spdx-3-model/blob/develop/model/AI/Properties/standardCompliance.md)
>   - Earlier "Standard" fields proposal https://github.com/spdx/spdx-3-model/issues/387
>   - StandardAndSpecification class -> consider renaming the class to Specification and add a new "spec status" property to tell if this is official standard, internal, etc.
>   - Consider adding Specification body property (currently utilising originatedBy property)

stevenc-stb commented:
Regulation class topic:
Mandatory with a regulation is unclear:
Products may have a mandatory regulation based on external factors
A common part used in multiple industries will have different implementation requirements - tire on a car as opposed to a plane
Mandatory within a region/industry 
Is Mandatory property of a regulation, property of a region or use industry?
Regulation processes?
Regulation action or compliance?
To be discussed at the Hardware Group.

bact commented:
The Hardware Profile group just kindly shared with us yesterday (2025-04-29 AI Team Meeting, see [tentative minutes here](https://spdx.swinslow.net/p/spdx-ai-minutes)) a diagram of Specfication class from their Hardware Profile proposal:
 
![Regulation as a subclass of Specification](https://github.com/user-attachments/assets/aa93cd1e-b0c8-4284-82ad-57fa4e9449b3)

bact commented:
@stevenc-stb I agree that "mandatory" is unclear. Another related terminology is "binding" and "non-binding". For me, if it is a property, it should be something intrinsic to the instance (a regulation). If it depends on jurisdiction or industry/sector, we may need to consider a relationship?

@colin-pm When I look again at "conformsTo" relationship type, I'm thinking about a proposal option from https://github.com/spdx/spdx-3-model/issues/893 that differentiated an aspiration to conform (during design/production) from the actual conformance (after certification), see below. How to capture this nuance?:

>    Currently, in the [AI BOM whitepaper](https://www.linuxfoundation.org/research/ai-bom) (p. 56), @bennetkl distinguished the two in this way:
>    - `standardName` - standards adhered to but that compliance was not obtained
>    - `standardCompliance` - standards that compliance are obtained (for example, from a third-party attestation or certification)

<img width="311" alt="standardName vs standardCompliance" src="https://github.com/user-attachments/assets/dee48b02-b518-4521-82df-9c16417536b7" />

bact commented:
Notes: We may also able to use or link with the Dublin Core terms "conformsTo" as well:
https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#http://purl.org/dc/terms/conformsTo

goneall commented:
Discussed on the tech call - we'll pick up the discussion on the 29 April 2025 tech call - first item on the agenda.

During the 22 April 2025 tech call we agreed this is a good use case to support, that it should be supported using a relationship, and we should have the same solution across multiple impacted profiles.

colin-pm commented:
I was thinking it over after the meeting today and I was somewhat inspired by the class diagram that I think Steven shared that was related to the one provided by @bact above.

An Attestation class or relationship should be what certifies that an Artifact meets a Regulation. The Attestation could be supported by evidence though a relationship of some form.

At that point, an artifact complying or conforming to a regulation could mean either aspirational or actual conformance. Attestation would be required to indicate that actual conformance is being met.

bact commented:
> An Attestation class or relationship should be what certifies that an Artifact meets a Regulation. The Attestation could be supported by evidence though a relationship of some form.

@colin-pm in 3.0.1 RelationshipType, there are two potential entries:

- hasDocumentation
- hasEvidence

There are also these relationship types:

- hasRequirement
- hasSpecification

colin-pm commented:
Sorry for the delay, I moved this PR's commit to the profile-hardware branch, updated Regulation to be a subclass of Specification, and removed the mandatory property from Regulation.

bact reviewed (approved):
Thank you @colin-pm  - LGTM

bact commented:
I think we can merge this to the profile-hardware branch now

goneall reviewed (approved):
LGTM - I'll go ahead and merge
